Source/Scene/ClippingPlanesCollection.js

```diff
@@ -0,0 +1,245 @@
+     * @param {Boolean} [options.enabled=true] Determines whether the clipping planes are active.
+     * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix specifying an additional transform relative to the clipping planes original coordinate system.
+     * @param {Boolean} [options.combineClippingRegions=true] If true, the region to be clipped must be included in all planes in this collection. Otherwise, a region will be clipped if included in any plane in the collection.
+     * @param {Color} [options.edgeColor=Color.WHITE] The color applied to highlight the edge along which an object is clipped.
+     * @param {Number} [options.edgeWidth=0.0] The width, in pixels, of the highlight applied to the edge along which an object is clipped.
+     */
+    function ClippingPlanesCollection(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * An array of up to 6 {@link Plane} objects used to selectively disable rendering on the outside of each plane.
+         *
+         * @type {Plane}
+         * @default []
+         */
+        this.planes = defaultValue(options.planes, []);
+
+        /**
+         * Determines whether the clipping planes are active.
+         *
+         * @type {Boolean}
+         * @default true
+         */
+        this.enabled = defaultValue(options.enabled, true);
+
+        /**
+         * The 4x4 transformation matrix specifying an additional transform relative to the clipping planes original coordinate system.
+         *
+         * @type {Matrix4}
+         * @default Matrix4.IDENTITY
+         */
+        this.modelMatrix = defaultValue(options.modelMatrix, Matrix4.clone(Matrix4.IDENTITY));
+
+        /**
+         * The color applied to highlight the edge along which an object is clipped.
+         *
+         * @type {Color}
+         * @default Color.WHITE
+         */
+        this.edgeColor = defaultValue(options.edgeColor, Color.clone(Color.WHITE));
+
+        /**
+         * The width, in pixels, of the highlight applied to the edge along which an object is clipped.
+         *
+         * @type {Number}
+         * @default 0.0
+         */
+        this.edgeWidth = defaultValue(options.edgeWidth, 0.0);
+
+        this._testIntersection = undefined;
+        this._combineClippingRegions = undefined;
+        this.combineClippingRegions = defaultValue(options.combineClippingRegions, true);
+    }
+
+    defineProperties(ClippingPlanesCollection.prototype, {
+        /**
+         * If true, the region to be clipped must be included in all planes in this collection.
+         * Otherwise, a region will be clipped if included in any plane in the collection.
+         *
+         * @memberof ClippingPlanesCollection.prototype
+         * @type {Boolean}
+         * @default true
+         */
+        combineClippingRegions : {
```

Comment:
I found the name of this property to be a little confusing and thought it did the opposite of what it actually does at first.  It's essentially either doing a `union` or an `intersection` of the regions, right?  The regions are more or less combined either way.  I would rename it to `unionClippingRegions` or something along those lines instead.